admin_manual/maintenance/backup.rst

```diff
@@ -63,32 +63,100 @@ Restoring Files From Backup When Encryption Is Enabled
 ------------------------------------------------------
 
 If you need to restore files from backup, which were backed up when encryption
-was enabled, here’s how to do it.
+was enabled, here's how to do it.
 
-.. NOTE:: 
+.. NOTE::
    This is effective from at least version v8.2.7 of ownCloud onwards. Also,
    this is **not officially supported**. ownCloud officially supports either
    restoring the full backup or restoring nothing — not restoring individual
    parts of it.
 
 1. Restore the file from backup.
-2. Restore the file's encryption keys from backup.
-3. Run ``occ files:scan``; this makes the scanner find it. Note that, in the DB
-   it will (1) have the "size" set to the encrypted size, which is wrong (and
-   bigger) and (2) the "encrypted" flag will be set to 0.
-4. Update the "encrypted" flag to 1 in the DB to all *files* under
-   ``files/path``, but **not** directories. Setting the flag to 1 tells the
-   encryption application that the file is encrypted and needs to be processed.
-   
+
+2. Restore the file's encryption keys from your backup.
+
+3. Run ``occ files:scan``; this makes the scanner find it.
+
+.. NOTE::
+   In the DB it will:
+
+   - Have the "size" set to the encrypted size, which is wrong (and bigger);
+   - The "encrypted" flag will be set to 0
+
+4. :ref:`Retrieve the encrypted flag value <retrieve_encrypted_flag_value_label>`.
+
+5. Update the encrypted flag.
+
 .. NOTE::
    There's no need to update the encrypted flag for files in either
    "files_versions" or "files_trashbin", because these aren't scanned or found
    by ``occ files:scan``.
-   
-5. Download the file once as the user; the file's size will be corrected
+
+6. Download the file once as the user; the file's size will be corrected
    automatically.
 
-This process might not be suitable across all environments. 
-If it’s not suitable for yours, you might need to run an OCC command that does
-the scanning. 
+This process might not be suitable across all environments.
+If it's not suitable for yours, you might need to run an OCC command that does the scanning.
 But, that will require the user's password or recovery key.
+
+.. _retrieve_encrypted_flag_value_label:
+
+Retrieve the Encrypted Flag Value
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. In the backup database, retrieve the ``numeric_id`` value for `the storage`_
+   where the file was located from the ``oc_storages`` table and store the value
+   for later reference.
+
+   For example, if you have the following in your ``oc_storages`` table, then
+   ``numeric_id`` will be ``2``.
+
+   ::
+
+    +--------------------------------+------------+-----------+--------------+
+    | id                             | numeric_id | available | last_checked |
+    +--------------------------------+------------+-----------+--------------+
+    | home::admin                    |          1 |         1 |         NULL |
+    | local::/var/www/owncloud/data/ |          2 |         1 |         NULL |
```

Comment:
hmmmm no, this storage is the root storage, don't use it for anything.

 I suggest you add more lines and one with "home::user1" which is the user we want to restore files for in our example

Reply:
Hmmm, ok.

Reply:
I'll have to read more on this.